main.tf

```diff
@@ -0,0 +1,31 @@
+### Add a user to the organization
+resource "github_membership" "membership_for_some_user" {
+  for_each = var.user_list
+  username = each.key
+  role     = each.value
+}
+
+### Add a team to the Organization
+resource "github_team" "some_team" {
+
+  for_each = var.add_teams
+  name     = each.key
+
+  description = "Some cool team"
+  privacy     = "closed"
+}
+
+### Add users from terraform.tfvars to their respective teams
+resource "github_team_members" "some_team_members" {
+
+  for_each = var.add_teams
+  team_id  = github_team.some_team[each.key].id
+
+  dynamic "members" {
+    for_each = var.add_teams[each.key].users
+    content {
+      username = members.value
+      role     = "member"
```

Comment:
How can we add a user with the role maintainer for this team?

Reply:
The task description specifies users to either be members or maintainers inside the organization, not the team. Should we also add this separation inside teams?